MoviesExample/app/src/main/java/com/novoda/demo/movies/MoviesDataSource.java

```diff
@@ -0,0 +1,77 @@
+package com.novoda.demo.movies;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.paging.PageKeyedDataSource;
+import android.support.annotation.NonNull;
+import android.util.Log;
+
+import com.novoda.demo.movies.api.MoviesResponse;
+import com.novoda.demo.movies.model.Movie;
+
+import java.util.List;
+
+public class MoviesDataSource extends PageKeyedDataSource<Integer, Movie> {
```

Comment:
The [PageKeyedDataSource](https://developer.android.com/reference/android/arch/paging/PageKeyedDataSource) is a [DataSource](https://developer.android.com/reference/android/arch/paging/DataSource) for paginated data.

There are also the [ItemKeyedDataSource](https://developer.android.com/reference/android/arch/paging/ItemKeyedDataSource) and [PositionalDataSource](https://developer.android.com/reference/android/arch/paging/PositionalDataSource) for other data architectures, for example like a fixed list.